src/System.Text.Primitives/System/Text/InvariantParser_int.cs

```diff
@@ -98,10 +88,18 @@ unsafe public static bool TryParse(byte* utf8Text, int index, int length, out in
             value = 0;
             bytesConsumed = 0;
             bool negative = false;
+            bool signed = false;
 
             if (utf8Text[index] == '-')
             {
                 negative = true;
+                signed = true;
```

Comment:
I think you might be missing the logic for determining the where the buffer ends.

    if (bytesConsumed == 0) // check to see if we've processed any digits at all
    {
           value = default(byte); // if we haven't, set value to 0 and return false
           return false;
    }

If no bytes were consumed, we return `false` (the user passed an invalid buffer). However, if one or more bytes were consumed, we assume that we were passed a valid number and return `true`.

Because of this logic, if the user passes a buffer of the form "-asfsd", the code will return `true` without the check you mentioned. The check you mentioned specifically looks and says "if we've consumed exactly one byte and that byte was a sign, then we just were directed to a + or a - sign in the middle of an invalid buffer (e.g. '-asdf'). No actual numbers were found, so we should return false."